lib/games_manager.rb

```diff
@@ -171,65 +171,43 @@ def get_season_games(season)
       game.game_id
     end
   end
-
+  
   def total_home_goals
-    home_team_id_goals = {}
-    games.each do |game|
-      if home_team_id_goals[game.home_team_id].nil?
-        home_team_id_goals[game.home_team_id] = game.home_goals
-      else
-        home_team_id_goals[game.home_team_id] += game.home_goals
-      end
+    team_id_by_home_goals = games.map do |game|
+      [game.home_team_id, game.home_goals]
     end
-    home_team_id_goals
+    sum_values(team_id_by_home_goals)
   end
-
-  def total_home_games
-    home_team_id_games = {}
-    games.each do |game|
-      if home_team_id_games[game.home_team_id].nil?
-        home_team_id_games[game.home_team_id] = 1
-      else
-        home_team_id_games[game.home_team_id] += 1
-      end
+  
+  def total_away_goals
+    team_id_by_away_goals = games.map do |game|
+      [game.away_team_id, game.away_goals]
     end
-    home_team_id_games
+    sum_values(team_id_by_away_goals)
   end
-
-  def highest_scoring_home
-    averages = total_home_goals.merge(total_home_games) do |team_id, goals, games|
-      (goals/games.to_f).round(2)
-    end
-    average_max = averages.max_by do |team_id, average|
-      average
+  
+  def total_home_games
+    team_id_by_home_games = games.map do |game|
+      [game.home_team_id, 1]
```

Comment:
i think this is a good approach. but if we iterate through the entire to get these value pairs, and then itterate through the list again to turn them into a hash, then we are asking the computer to do the work twice. we can probobly afford the extra time, but we should conider whether it would be possible to iterate through the list once, and get the pairs we need that way?

Reply:
You 're correct we're iterating through twice but the benefit is creating code that is more reusable. I compared the time it took to run the rake as well as the time it took to run the spec harness and the difference was significantly less than one second in both cases.